backend/src/middleware/handleHtmlContent/notifications/extractMentionedUsers.spec.js

```diff
@@ -1,59 +1,30 @@
+const contentEmptyMentions =
+  '<p>Something inspirational about <a href="/profile/u2" data-mention-id="" target="_blank">@bob-der-baumeister</a> and <a href="/profile/u3/jenny-rostock" class="mention" data-mention-id target="_blank">@jenny-rostock</a>.</p>'
+const contentWithPlainLinks =
+  '<p>Something inspirational about <a class="mention" href="/profile/u2" target="_blank">@bob-der-baumeister</a> and <a href="/profile/u3" target="_blank">@jenny-rostock</a>.</p>'
+
 describe('extractMentionedUsers', () => {
   describe('content undefined', () => {
     it('returns empty array', () => {
       expect(extractMentionedUsers()).toEqual([])
     })
   })
 
-  describe('searches through links', () => {
-    it('ignores links without .mention class', () => {
-      const content =
-        '<p>Something inspirational about <a href="/profile/u2" target="_blank">@bob-der-baumeister</a> and <a href="/profile/u3" target="_blank">@jenny-rostock</a>.</p>'
-      expect(extractMentionedUsers(content)).toEqual([])
-    })
-
-    describe('given a link with .mention class', () => {
-      it('extracts ids', () => {
-        const content =
-          '<p>Something inspirational about <a href="/profile/u2" class="mention" target="_blank">@bob-der-baumeister</a> and <a href="/profile/u3/jenny-rostock" class="mention" target="_blank">@jenny-rostock</a>.</p>'
-        expect(extractMentionedUsers(content)).toEqual(['u2', 'u3'])
-      })
-
-      describe('handles links', () => {
-        it('with slug and id', () => {
-          const content =
-            '<p>Something inspirational about <a href="/profile/u2/bob-der-baumeister" class="mention" target="_blank">@bob-der-baumeister</a> and <a href="/profile/u3/jenny-rostock/" class="mention" target="_blank">@jenny-rostock</a>.</p>'
-          expect(extractMentionedUsers(content)).toEqual(['u2', 'u3'])
-        })
-
-        it('with domains', () => {
-          const content =
-            '<p>Something inspirational about <a href="http://localhost:3000/profile/u2/bob-der-baumeister" class="mention" target="_blank">@bob-der-baumeister</a> and <a href="http://localhost:3000//profile/u3/jenny-rostock/" class="mention" target="_blank">@jenny-rostock</a>.</p>'
-          expect(extractMentionedUsers(content)).toEqual(['u2', 'u3'])
-        })
-
-        it('special characters', () => {
-          const content =
-            '<p>Something inspirational about <a href="http://localhost:3000/profile/u!*(),2/bob-der-baumeister" class="mention" target="_blank">@bob-der-baumeister</a> and <a href="http://localhost:3000//profile/u.~-3/jenny-rostock/" class="mention" target="_blank">@jenny-rostock</a>.</p>'
-          expect(extractMentionedUsers(content)).toEqual(['u!*(),2', 'u.~-3'])
-        })
-      })
+  it('ignores links without .mention class', () => {
+    expect(extractMentionedUsers(contentWithPlainLinks)).toEqual([])
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-08-01T05:46:22Z" title="Thursday, August 1st 2019, 7:46:22 am +02:00">Aug 1, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -1,59 +1,30 @@
 import extractMentionedUsers from './extractMentionedUsers'
 
+const contentWithMentions =
+  '<p>Something inspirational about <a href="/profile/u2" class="not-a-mention" data-mention-id="bobs-id" target="_blank">@bob-der-baumeister</a> and <a href="/profile/u3/jenny-rostock" class="mention" data-mention-id="u3" target="_blank">@jenny-rostock</a>.</p>'
+const contentEmptyMentions =
+  '<p>Something inspirational about <a href="/profile/u2" data-mention-id="" target="_blank">@bob-der-baumeister</a> and <a href="/profile/u3/jenny-rostock" class="mention" data-mention-id target="_blank">@jenny-rostock</a>.</p>'
+const contentWithLinks =
+  '<p>Something inspirational about <a class="mention" href="/profile/u2" target="_blank">@bob-der-baumeister</a> and <a href="/profile/u3" target="_blank">@jenny-rostock</a>.</p>'
+
 describe('extractMentionedUsers', () => {
   describe('content undefined', () => {
     it('returns empty array', () => {
       expect(extractMentionedUsers()).toEqual([])
     })
   })
 
-  describe('searches through links', () => {
-    it('ignores links without .mention class', () => {
-      const content =
-        '<p>Something inspirational about <a href="/profile/u2" target="_blank">@bob-der-baumeister</a> and <a href="/profile/u3" target="_blank">@jenny-rostock</a>.</p>'
-      expect(extractMentionedUsers(content)).toEqual([])
-    })
-
-    describe('given a link with .mention class', () => {
-      it('extracts ids', () => {
-        const content =
-          '<p>Something inspirational about <a href="/profile/u2" class="mention" target="_blank">@bob-der-baumeister</a> and <a href="/profile/u3/jenny-rostock" class="mention" target="_blank">@jenny-rostock</a>.</p>'
-        expect(extractMentionedUsers(content)).toEqual(['u2', 'u3'])
-      })
-
-      describe('handles links', () => {
-        it('with slug and id', () => {
-          const content =
-            '<p>Something inspirational about <a href="/profile/u2/bob-der-baumeister" class="mention" target="_blank">@bob-der-baumeister</a> and <a href="/profile/u3/jenny-rostock/" class="mention" target="_blank">@jenny-rostock</a>.</p>'
-          expect(extractMentionedUsers(content)).toEqual(['u2', 'u3'])
-        })
-
-        it('with domains', () => {
-          const content =
-            '<p>Something inspirational about <a href="http://localhost:3000/profile/u2/bob-der-baumeister" class="mention" target="_blank">@bob-der-baumeister</a> and <a href="http://localhost:3000//profile/u3/jenny-rostock/" class="mention" target="_blank">@jenny-rostock</a>.</p>'
-          expect(extractMentionedUsers(content)).toEqual(['u2', 'u3'])
-        })
-
-        it('special characters', () => {
-          const content =
-            '<p>Something inspirational about <a href="http://localhost:3000/profile/u!*(),2/bob-der-baumeister" class="mention" target="_blank">@bob-der-baumeister</a> and <a href="http://localhost:3000//profile/u.~-3/jenny-rostock/" class="mention" target="_blank">@jenny-rostock</a>.</p>'
-          expect(extractMentionedUsers(content)).toEqual(['u!*(),2', 'u.~-3'])
-        })
-      })
+  it('ignores links without .mention class', () => {
+    expect(extractMentionedUsers(contentWithLinks)).toEqual([])
```


`contentWithLinks` does contain the `mention` class, Oder?
I think what it is missing is the `data-mention-id`(?)

Reply:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-08-01T10:11:03Z" title="Thursday, August 1st 2019, 12:11:03 pm +02:00">Aug 1, 2019</time>_

---


I renamed the variable to `contentWithPlainLinks`, it should ignore the links if they don't have `data-mention-id` . This is unambiguous, oder?

Reply:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-08-01T10:28:18Z" title="Thursday, August 1st 2019, 12:28:18 pm +02:00">Aug 1, 2019</time>_

---


sorry, what I was questioning is the description `ignores links without .mention class`, cause the `contentWithPlainLinks` does include a mention class, but it is ignored for the reason you stated :point_up:

Reply:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-08-01T10:40:29Z" title="Thursday, August 1st 2019, 12:40:29 pm +02:00">Aug 1, 2019</time>_

---


I'm stupid, I meant the `data-mention-id`